src/libraries/Common/src/Interop/Unix/System.Globalization.Native/Interop.Normalization.cs

```diff
@@ -12,7 +13,10 @@ internal static partial class Globalization
         [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_IsNormalized")]
         internal static extern int IsNormalized(NormalizationForm normalizationForm, string src, int srcLen);
 
+        internal static int NormalizeString(NormalizationForm normalizationForm, string src, int srcLen, Span<char> dstBuffer) =>
```

Comment:
>string [](start = 81, length = 6)

do we need this? does calling the overload with span length is big deal?

Reply:
It now also does pinning of inputs, providing more utility. Is that okay or should that be moved to call sites instead?

Reply:
usually, we do the pinning in the calling sites.

Reply:
Changed